tensorboard/plugins/hparams/backend_context.py

```diff
@@ -191,8 +191,8 @@ def read_last_scalars(self, ctx, experiment_id, run_tag_filter):
             plugin_name=scalar_metadata.PLUGIN_NAME,
             run_tag_filter=run_tag_filter,
             # TODO(#3436): We assume that downsampling always includes
-            # the most recent datum, which holds for all implementations
-            # of interest but is not yet required by the interface.
+            # the most recent datum for some data provider implementations
+            # of interest. This is not required by the interface.
             downsample=1,
```

Comment:
Do not all existing "implementations of interest" do this anymore? If not, it feels odd that only some implementations meet this "contract", but we're still using the same code even if they don't.

Is this still a TODO (assuming the idea is to make it "required" by the interface somehow, maybe specifying a down-sampling strategy)?

Reply:
Great point! I think it makes sense to actually update the `read_scalars` contract to always include the last scalar. Removed this TODO instead.